src/codecs/format/bytes.rs

```diff
@@ -47,13 +51,65 @@ impl Deserializer for BytesDeserializer {
     }
 }
 
+/// Config used to build a `BytesSerializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct BytesSerializerConfig;
+
+impl BytesSerializerConfig {
+    /// Creates a new `BytesSerializerConfig`.
+    pub const fn new() -> Self {
+        Self
+    }
+}
+
+#[typetag::serde(name = "bytes")]
+impl SerializerConfig for BytesSerializerConfig {
+    fn build(&self) -> crate::Result<BoxedSerializer> {
+        Ok(Box::new(BytesSerializer))
+    }
+}
+
+/// Serializer that converts bytes to an `Event`.
+///
+/// This serializer can be considered as the no-op action for input where no
+/// further encoding has been specified.
```

Comment:
We've caused a decent amount of confusion in the past by considering the message field as a "no-op" or default option for encoding, so I'd rather not replicate that here. Two points on this:

1. I'm not sure it makes sense to have a "no-op" encoder. There fundamentally has to be some operation to turn an event into bytes.
2. A serializer that just takes the message field as-is should be named in way that makes that clear.

Reply:
It refers to that behavior more explicitly now:

https://github.com/vectordotdev/vector/blob/04c368e8fe8e06238c1cdc9ef1e7c8836029a1b6/src/codecs/format/text.rs#L28-L30